src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -324,13 +348,22 @@ object Ledger {
   type PR = ProgramResult[InMemoryWorldStateProxy, InMemoryWorldStateProxyStorage]
 
   case class BlockResult(worldState: InMemoryWorldStateProxy, gasUsed: BigInt = 0, receipts: Seq[Receipt] = Nil)
+  case class BlockPreparationResult(blockResult: BlockResult, stateRootHash: ByteString)
   case class TxResult(worldState: InMemoryWorldStateProxy, gasUsed: BigInt, logs: Seq[TxLogEntry])
 }
 
-sealed trait BlockExecutionError
+sealed trait BlockExecutionError{
+  val reason: String
+}
 
 object BlockExecutionError {
   case class ValidationBeforeExecError(reason: String) extends BlockExecutionError
   case class TxsExecutionError(reason: String) extends BlockExecutionError
   case class ValidationAfterExecError(reason: String) extends BlockExecutionError
 }
+
+trait BlockPreparationError
+
+object BlockPreparationError {
+  case class TxError(reason: String) extends BlockPreparationError
```

Comment:
Was it the intention to group all `BlockPreparationError`s? There are others in `BlockGenerator`